server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2114,9 +2114,13 @@ public void deletePool(Pool pool) {
         Event event = eventFactory.poolDeleted(pool);
 
         // Must do a full revoke for all entitlements:
-        for (Entitlement e : poolCurator.entitlementsIn(pool)) {
+        List<Entitlement> entitlements = poolCurator.entitlementsIn(pool);
+        for (Entitlement e : entitlements) {
             e.setDeletedFromPool(true);
-            revokeEntitlement(e);
+        }
+
+        for (List<Entitlement> partition : Iterables.partition(entitlements, 1000)) {
+            revokeEntitlements(partition);
```

Comment:
As per our conversation, there's more things to discuss here. The ideal solution is to not fetch the entitlement objects into memory at all (they're 60k+ after all), but instead do a batch delete by entitlement IDs like is done elsewhere.
To do that though, we first have to make sure we do anything else that needs to be done (revokeEntitlements does a lot of stuff, and we need to find out if we even need to do most of those in this specific use case (UndoImportsJob)).